src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,63 @@
-export const PeopleFilters = () => {
+import { Link, useSearchParams } from 'react-router-dom';
+import React, { ChangeEvent } from 'react';
+import { getSearchWith, SearchParams } from '../utils/searchHelper';
+import classNames from 'classnames';
+
+const SEX_FILTERS = [
+  { label: 'All', value: null },
+  { label: 'Male', value: 'm' },
+  { label: 'Female', value: 'f' },
+];
+
+const CENTURIES = ['16', '17', '18', '19', '20'];
+
+export const PeopleFilters: React.FC = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const handleSetParams = (newParams: SearchParams) => {
+    const updatedParams = getSearchWith(searchParams, newParams);
+
+    setSearchParams(updatedParams);
+  };
+
+  const addCenturyParams = (century: string) => {
+    const centuries = searchParams.getAll('century');
+
+    handleSetParams({
+      century: centuries.includes(century)
+        ? centuries.filter(c => c !== century)
+        : [...centuries, century],
+    });
+  };
+
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex') || null;
+
+  const handleQueryChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const newQuery = e.target.value || null;
+
+    handleSetParams({ query: newQuery });
+  };
+
   return (
     <nav className="panel">
       <p className="panel-heading">Filters</p>
 
       <p className="panel-tabs" data-cy="SexFilter">
-        <a className="is-active" href="#/people">
-          All
-        </a>
-        <a className="" href="#/people?sex=m">
-          Male
-        </a>
-        <a className="" href="#/people?sex=f">
-          Female
-        </a>
+        {SEX_FILTERS.map(filter => (
+          <Link
```

Comment:
You can use [SearchLink](https://github.com/LiliiaVol/react_people-table-advanced/blob/develop/src/components/SearchLink.tsx) that does this work for you, but it's okay